internal/server/admin/keyset_service.go

```diff
@@ -0,0 +1,145 @@
+package admin
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	pbModel "github.com/kong/koko/internal/gen/grpc/kong/admin/model/v1"
+	v1 "github.com/kong/koko/internal/gen/grpc/kong/admin/service/v1"
+	"github.com/kong/koko/internal/model"
+	"github.com/kong/koko/internal/resource"
+	"github.com/kong/koko/internal/server/util"
+	"github.com/kong/koko/internal/store"
+	"go.uber.org/zap"
+)
+
+type KeySetService struct {
+	v1.UnimplementedKeySetServiceServer
+	CommonOpts
+}
+
+func (s *KeySetService) GetKeySet(
+	ctx context.Context,
+	req *v1.GetKeySetRequest,
+) (*v1.GetKeySetResponse, error) {
+	logger := s.logger(ctx)
+	db, err := s.getDB(ctx, req.Cluster)
+	if err != nil {
+		return nil, err
+	}
+	result := resource.NewKeySet()
+	err = getEntityByIDOrName(ctx, req.Id, result, store.GetByName(req.Id), db, logger)
+	if err != nil {
+		return nil, util.HandleErr(ctx, logger, err)
+	}
+
+	return &v1.GetKeySetResponse{
+		Item: result.KeySet,
+	}, nil
+}
+
+func (s *KeySetService) CreateKeySet(
+	ctx context.Context,
+	req *v1.CreateKeySetRequest,
+) (*v1.CreateKeySetResponse, error) {
+	db, err := s.getDB(ctx, req.Cluster)
+	if err != nil {
+		return nil, err
+	}
+
+	res := resource.NewKeySet()
+	res.KeySet = req.Item
+	s.logger(ctx).Debug("copied keyset", zap.Any("keyset", res.KeySet))
+	if err := db.Create(ctx, res); err != nil {
+		s.logger(ctx).Error("error creating", zap.Error(err))
```

Comment:
Can we have this be a full sentence? e.g.:

```suggestion
		s.logger(ctx).Error("unable to create keyset entity", zap.Error(err))
```